src/plugins/intel_cpu/src/emitters/snippets/riscv64/jit_loop_emitters.cpp

```diff
@@ -0,0 +1,307 @@
+                                          const std::vector<size_t>& out,
+                                          const std::vector<size_t>& pool_vec_idxs,
+                                          const std::vector<size_t>& pool_gpr_idxs) const {
+    validate_arguments(in, out);
+    ov::intel_cpu::riscv64::jit_emitter::emit_code_impl(in, out, pool_vec_idxs, pool_gpr_idxs, {});
+}
+
+void jit_loop_end_emitter::emit_impl(const std::vector<size_t>& in,
+                                     [[maybe_unused]] const std::vector<size_t>& out) const {
+    // Build list of data pointer regs: in[0..io_size-1], work_amount is in.back()
+    std::vector<size_t> data_ptr_reg_idxs;
+    const size_t io_size = num_inputs + num_outputs;
+    data_ptr_reg_idxs.reserve(io_size);
+    std::copy(in.begin(), in.end() - 1, std::back_inserter(data_ptr_reg_idxs));
+
+    auto apply_increments =
+        [&](bool use_runtime_args, size_t field_offset, const std::vector<int64_t>& increments, size_t scale) {
+            // Acquire two scratch regs
+            std::vector<size_t> used = in;  // exclude live inputs
+            jit_aux_gpr_holder h_incs(h, aux_gpr_idxs, used);
+            jit_aux_gpr_holder h_tmp(h, aux_gpr_idxs, used);
+            Xbyak_riscv::Reg reg_increments = h_incs.get_reg();
+            Xbyak_riscv::Reg tmp = h_tmp.get_reg();
+            auto add_increments = [&]() {
+                for (size_t idx = 0; idx < data_ptr_reg_idxs.size(); ++idx) {
+                    const auto& inc = increments[idx];
+                    if (is_incremented[idx] && inc != 0) {
+                        auto ptr_reg = Xbyak_riscv::Reg(data_ptr_reg_idxs[idx]);
+                        if (ov::snippets::utils::is_dynamic_value(inc)) {
+                            // ptr += ((int64*)reg_increments)[idx]
+                            h->uni_li(tmp, idx * sizeof(int64_t));
+                            h->add(tmp, reg_increments, tmp);
+                            // reuse tmp as inc_val after load
+                            h->ld(tmp, tmp, 0);
+                            // inc_val is in elements; convert to bytes = inc_val * scale * data_sizes[idx]
+                            size_t mul = scale * static_cast<size_t>(data_sizes[idx]);
+                            if (mul != 1) {
+                                jit_aux_gpr_holder h_mul(h, aux_gpr_idxs, used);
+                                auto mulreg = h_mul.get_reg();
+                                h->uni_li(mulreg, mul);
+                                h->mul(tmp, tmp, mulreg);
+                            }
+                            h->add(ptr_reg, ptr_reg, tmp);
+                        } else {
+                            size_t add_bytes = static_cast<size_t>(inc) * scale * static_cast<size_t>(data_sizes[idx]);
+                            h->uni_li(tmp, add_bytes);
+                            h->add(ptr_reg, ptr_reg, tmp);
+                        }
```

Comment:
Currently, ptr increments are represented differently for static and dynamic cases:
- in static case, we should multiply the increments by  scale and data_sizes[idx]
- in dynamic case, scale and data_sizes[idx] are already taken into account in `loop_args_t`

Can we eliminate this representation misalignment? Let's store loop_args as emitter member, and form it in the constructor (as it is done in `jit_loop_end_base_emitter::compose_loop_args`). Probably, we can even move `compose_loop_args` to some common place and reuse it here